src/sagas/RootSaga.js

```diff
@@ -66,35 +69,60 @@ export function* handleFetchProfile(action) {
 }
 
 export function* handleSaveProfile(action) {
-  const { username, accountDrafts, visibilityDrafts } = yield select(handleSaveProfileSelector);
-
   try {
+    const { username, drafts, preferences } = yield select(handleSaveProfileSelector);
+
+    const accountDrafts = keepKeys(drafts, [
+      'bio',
+      'courseCertificates',
+      'country',
+      'education',
+      'languageProficiencies',
+      'name',
+      'socialLinks',
+    ]);
+
+    const preferencesDrafts = keepKeys(drafts, [
+      'visibilityBio',
+      'visibilityCourseCertificates',
+      'visibilityCountry',
+      'visibilityEducation',
+      'visibilityLanguageProficiencies',
+      'visibilityName',
+      'visibilitySocialLinks',
+    ]);
+
+    if (Object.keys(preferencesDrafts).length > 0) {
+      preferencesDrafts.accountPrivacy = 'custom';
```

Comment:
This is a special little nugget.  If we've updated any visibilities at all, we set account privacy to 'custom'.